includes/admin/class-wc-rest-payments-settings-controller.php

```diff
@@ -763,21 +768,58 @@ private function update_platform_checkout_enabled_locations( WP_REST_Request $re
 	}
 
 	/**
-	 * Updates the settings of advanced fraud protection rules.
+	 * Updates the settings of fraud protection rules (both settings and level in one function, because they are connected).
 	 *
 	 * @param WP_REST_Request $request Request object.
 	 */
-	private function update_advanced_fraud_protection_settings( WP_REST_Request $request ) {
+	private function update_fraud_protection_settings( WP_REST_Request $request ) {
 		if ( ! WC_Payments_Features::is_fraud_protection_settings_enabled() ) {
 			return;
 		}
 
-		if ( ! $request->has_param( 'advanced_fraud_protection_settings' ) ) {
+		if ( ! $request->has_param( 'current_protection_level' ) || ! $request->has_param( 'advanced_fraud_protection_settings' ) ) {
+			return;
+		}
+
+		$protection_level = $request->get_param( 'current_protection_level' );
+
+		// Check validity of the protection level.
+		if ( ! in_array( $protection_level, [ 'basic', 'standard', 'high', 'advanced' ], true ) ) {
 			return;
 		}
 
-		$advanced_fraud_protection_settings = $request->get_param( 'advanced_fraud_protection_settings' );
+		// Get rulesets per protection level.
+		switch ( $protection_level ) {
+			case 'basic':
+				$ruleset_config = Fraud_Risk_Tools::get_basic_protection_settings();
+				break;
+			case 'standard':
+				$ruleset_config = Fraud_Risk_Tools::get_standard_protection_settings();
+				break;
+			case 'high':
+				$ruleset_config = Fraud_Risk_Tools::get_high_protection_settings();
+				break;
+			case 'advanced':
+				$referer                   = $request->get_header( 'referer' );
+				$is_advanced_settings_page = 0 < strpos( $referer, 'fraud-protection' );
+				if ( $is_advanced_settings_page ) {
```

Comment:
This if statement seems to be inverted according to the comments.

When we change the protection levels to **Basic -> Standard -> Advanced** without refreshing the page, it changes successfully to **Standard**. After saving the **Advanced**, it is reset to **Basic**.

It works as expected If we change it to `if ( ! $is_advanced_settings_page )`.

Reply:
Thank you for catching this and letting me free from some time of digging deep! Yes, that is the culprit that causes it!

Reply:
Fixed in [9b2dd0a](https://github.com/Automattic/woocommerce-payments/pull/5685/commits/9b2dd0a842f7f829e973ab3b6d264edd14f1eb91).